python/cudf/cudf/options.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional, Union
+
+
+@dataclass
+class CUDFOption:
+    value: Any
+    description: str
+    validator: Callable
+
+
+_CUDF_OPTIONS: Dict[str, CUDFOption] = {}
+
+
+def _register_option(
+    name: str, default_value: Any, description: str, validator: Callable
+):
+    """Register an entry in the option dictionary.
+
+    Parameters
+    ----------
+    name : str
+        The name of the option. Also used as the key in the dictionary.
+
+    default_value : Any
+        The default value of the option.
+
+    description : str
+        A text description of the option.
+
+    validator : Callable
+        A function that returns ``True`` if a given value is valid for the
+        option, ``False`` otherwise.
+    """
+    if not validator(default_value):
+        raise ValueError(f"Invalid default value: {default_value}")
+
+    _CUDF_OPTIONS[name] = CUDFOption(default_value, description, validator)
+
+
+def get_option(key: str) -> Any:
```

Comment:
Probably want to match "name" above and in pandas.
```suggestion
def get_option(name: str) -> Any:
```